tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -85,11 +117,31 @@ def index_impl(self):
 
   def scalars_impl(self, tag, run, output_format):
     """Result of the form `(body, mime_type)`."""
-    tensor_events = self._multiplexer.Tensors(run, tag)
-    values = [[tensor_event.wall_time,
-               tensor_event.step,
-               tf.make_ndarray(tensor_event.tensor_proto).item()]
-              for tensor_event in tensor_events]
+    if self._db_connection_provider:
+      db = self._db_connection_provider()
+      cursor = db.execute(
+          ('SELECT '
+           'Tensors.computed_time, '
+           'Tensors.data, '
+           'Tensors.step, '
```

Comment:
nit: makes sense to keep "step" and "computed_time" adjacent and "data" and "dtype" adjacent given that the former two are metadata and the second two define the actual tensor itself?

Also, I wonder if it's worth validating the shape field somewhere just in case somebody writes a non-scalar tensor and applies the scalars plugin tag inadvertently.  Maybe just a `WHERE Tensors.shape = ''`?

Reply:
+1 to both. Done.